File: problem-3/author-solutions/s.cpp
#include<iostream>
#include<vector>
using namespace std;
bool isBalanced(long long x) {
  string s = to_string(x);
  int n = s.size();
  int n2 = n/2;
  int leftSum = 0, rightSum = 0;
  for (int i = 0; i < n2; i++) {
    leftSum += s[i] - '0';
    rightSum += s[n -1 -i] - '0';
  }
  return leftSum == rightSum;
}
int main(void) {
  int n; cin >> n;
  vector<int> v(n);
  vector<long long> p(n);
  cin >> v[0];
  p[0] = v[0];
  for (int i=1; i<n; i++) {
    cin >> v[i];
    p[i] = p[i - 1] + v[i];
  }
  int count = 0;
  for (int size = 1; size <= n; size++) {
    int last = n - size + 1;
    for (int start = 0; start < last; start++) {
      int end = start + size - 1;
      long long sum = p[end];
      if (start != 0) sum -= p[start - 1];
      // cout << size << ' ' << start << ' '<< end << ' ' << sum << ' ' << isBalanced(sum) << '\n';
      if (isBalanced(sum)) count++;
    }
  }
  cout << count;
  return 0;
}

File: problem-3/author-solutions/s.py
def isBalanced(x):
  s = str(x)
  n = len(s)
  n2 = n//2
  leftSum = rightSum = 0
  for i in range(n2):
    leftSum += ord(s[i]) - ord('0')
    rightSum += ord(s[n -1 -i]) - ord('0')
  return leftSum == rightSum

n = int(input())
v = [int(x) for x in input().split()]
p = [0] * n
p[0] = v[0]
for i in range(1, n):
  p[i] = p[i - 1] + v[i]
count = 0
for size in range(1, n+1):
  last = n - size + 1
  for start in range(last):
    end = start + size - 1
    sum = p[end]
    if start != 0:
      sum -= p[start - 1]
    if (isBalanced(sum)):
      count += 1
print(count)